src/main/java/com/aspectsecurity/contrast/contrastjenkins/VulnerabilityTrendHelper.java

```diff
@@ -6,22 +6,41 @@
 import com.contrastsecurity.models.AgentType;
 import com.contrastsecurity.models.Rules;
 import com.contrastsecurity.sdk.ContrastSDK;
+import hudson.ProxyConfiguration;
 import hudson.model.Run;
 import hudson.model.TaskListener;
 import hudson.util.ListBoxModel;
+import jenkins.model.Jenkins;
 import org.apache.commons.lang3.StringUtils;
 
 import java.io.IOException;
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Collections;
-import java.util.EnumSet;
-import java.util.List;
+import java.net.MalformedURLException;
+import java.net.Proxy;
+import java.net.URL;
+import java.util.*;
 
 public class VulnerabilityTrendHelper {
 
     public static ContrastSDK createSDK(String username, String serviceKey, String apiKey, String teamServerUrl) {
-        return new ContrastSDK(username, serviceKey, apiKey, teamServerUrl);
+        ContrastSDK contrastSDK = null;
+        ProxyConfiguration proxyConfig = Jenkins.getInstance().proxy;
+        URL url = null;
+        try {
+            url = new URL(teamServerUrl);
+        } catch (MalformedURLException e) {
+            e.printStackTrace();
```

Comment:
We shouldn't just stack trace this guy and move on. If we're hitting this exception, we're done. We can't contact teamserver. 

Log and error to the (Jenkins?) console and throw the exception.

Reply:
I think it may be better if treat the malformed url as just a wrong url. Plugin already handles errors for such case.